File: test/Data.Analyzers.Vsix.Test/CSharp/PrimaryKeyMismatchSortAttribute.cs
using DevZest.Data.Annotations;

namespace DevZest.Data.Analyzers.Vsix.Test.CSharp
{
    public sealed class PrimaryKeyMismatchSortAttribute : CandidateKey
    {
        public PrimaryKeyMismatchSortAttribute([Asc]_Int32 id1, _Int32 id2)
            : base(id1, id2.Desc())
        {
        }
    }
}

File: test/Data.Analyzers.Vsix.Test/VisualBasic/PrimaryKeyMismatchSortAttribute.vb
Imports DevZest.Data.Annotations

Public NotInheritable Class PrimaryKeyMismatchSortAttribute
    Inherits PrimaryKey

    Public Sub New(<Asc()> id1 As _Int32, id2 As _Int32)
        MyBase.New(id1, id2.Desc())
    End Sub

    Public ReadOnly Property ID1 As _Int32
        Get
            Return GetColumn(Of _Int32)(0)
        End Get
    End Property

    Public ReadOnly Property ID2 As _Int32
        Get
            Return GetColumn(Of _Int32)(1)
        End Get
    End Property

End Class
